app/cdap/components/WrangleHome/Components/OngoingDataExplorations/utils.ts

```diff
@@ -59,49 +59,52 @@ export const getUpdatedExplorationCards = (
     Array.isArray(existingExplorationCards) &&
     existingExplorationCards.length
   ) {
-    existingExplorationCards.forEach((eachExplorationCard) => {
-      const eachExplorationCardData = [];
-      Object.keys(eachExplorationCard).map((keys) => {
-        const onGoingDatExplorationData = {} as IOnGoingDataExplorationsData;
-        switch (keys) {
-          case CONNECTOR_TYPE:
-            onGoingDatExplorationData.icon = getIconForConnector(eachExplorationCard[keys]);
-            onGoingDatExplorationData.label = eachExplorationCard[keys];
-            onGoingDatExplorationData.type = ICON;
-            break;
-          case CONNECTION_NAME:
-            onGoingDatExplorationData.label = eachExplorationCard[keys];
-            onGoingDatExplorationData.type = ICON_WITH_TEXT;
-            break;
-          case WORKPSACE_NAME:
-            onGoingDatExplorationData.label = eachExplorationCard[keys];
-            onGoingDatExplorationData.type = TEXT;
-            break;
-          case RECIPE_STEPS_KEY:
-            onGoingDatExplorationData.label = [
-              eachExplorationCard[keys],
-              T.translate(`${PREFIX}.recipeSteps`),
-            ].join(' ');
-            onGoingDatExplorationData.type = TEXT;
-            break;
-          case DATA_QUALITY:
-            onGoingDatExplorationData.label = Number(eachExplorationCard[keys]);
-            onGoingDatExplorationData.percentageSymbol = '%';
-            onGoingDatExplorationData.subText = T.translate(`${PREFIX}.nullValues`);
-            onGoingDatExplorationData.type = PERCENTAGE_WITH_TEXT;
-            break;
-          case WORKSPACE_ID:
-            onGoingDatExplorationData.workspaceId = eachExplorationCard[keys];
-            break;
-          case COUNT:
-            onGoingDatExplorationData.count = eachExplorationCard[keys];
-            break;
-        }
-        eachExplorationCardData.push(onGoingDatExplorationData);
-      });
+    existingExplorationCards
+      .filter((eachItem) => eachItem.count !== 0)
+      .filter((_, itemIndex) => (cardCount && itemIndex < cardCount) || !cardCount)
+      .forEach((eachExplorationCard) => {
+        const eachExplorationCardData = [];
+        Object.keys(eachExplorationCard).map((keys) => {
+          const onGoingDatExplorationData = {} as IOnGoingDataExplorationsData;
+          switch (keys) {
```

Comment:
shall we move this switch to a separate util function called getUpdatedOngoingExplorationData and return updated value for reassignment
plz check if it makes sense